pkg/test/framework/resource/cluster.go

```diff
@@ -58,6 +58,15 @@ func (c Clusters) Names() []string {
 	return names
 }
 
+// ByNetwork returns a map of network name to a subset of clusters
+func (c Clusters) ByNetwork() map[string]Clusters {
```

Comment:
We should try to fold the accessors for control plane and network topologies into `Cluseters` and get rid of the methods on Environment.